mora/service/common.py

```diff
@@ -26,6 +27,16 @@ class FieldTypes(Enum):
     ADAPTED_ZERO_TO_MANY = 2,
 
 
+class AbstractRelationDetail(abc.ABC):
```

Comment:
Overall, I like this idea.
I'm guessing the end goal is having separate strategies for each type of detail, i.e. engagement, association etc.?

Reply:
Generally speaking, the point is that IT systems and addresses are similar, yet different from the others. Eventually, yes, it might make sense to have each service as a simple class.

Reply:
Okay, good. I might be trying to plan a bit too far ahead for when we potentially have to support a number of slightly different implementations for each municipality -- but I'd really like, sooner rather than later, to extract all of the specific logic, in an open/closed fashion to make it as easy as possible to swap out all the specific implementation details. This is a good step in that direction at least.

Reply:
Yeah, we're in agreement here — we want to keep moving slowly but steadily in that direction 🙂